corehq/apps/app_manager/static/app_manager/js/app_view_application.js

```diff
@@ -61,5 +66,42 @@ hqDefine("app_manager/js/app_view_application", function() {
                 $form.submit();
             }
         });
+
+        var validateCopyApplicationForm = function(form){
+            var domainDiv  = form.find("#div_id_domain"),
+                appNameDiv = form.find("#div_id_name"),
+                domain = domainDiv.find("#id_domain"),
+                appName = appNameDiv.find("#id_name"),
+                error = false,
+                domainNames = initial_page_data("domain_names");
+
+            appNameDiv.removeClass('has-error');
+            domainDiv.removeClass('has-error');
+            form.find('.help-block').remove();
```

Comment:
If you turn on the feature flag for linking domains, you'll see this additional checkbox and help text, and the help text is getting removed if there's an error in the form.

<img width="728" alt="screen shot 2018-08-08 at 11 04 25 am" src="https://user-images.githubusercontent.com/1486591/43845817-d980be62-9afa-11e8-88fd-9c1263ec77ef.png">